src/simtools/production_configuration/scale_events_manager.py

```diff
@@ -0,0 +1,117 @@
+        self.output_filepath = Path(self.output_path).joinpath(f"{self.args['output_file']}")
+        self.metrics = collect_data_from_file(self.args["metrics_file"])
+        self.evaluator_instances = []
+
+    def initialize_evaluators(self):
+        """Initialize StatisticalErrorEvaluator instances for the given zeniths and offsets."""
+        if not (self.args["base_path"] and self.args["zeniths"] and self.args["offsets"]):
+            self.logger.warning("No files read")
+            self.logger.warning(f"Base Path: {self.args['base_path']}")
+            self.logger.warning(f"Zeniths: {self.args['zeniths']}")
+            self.logger.warning(f"Offsets: {self.args['offsets']}")
+            return
+
+        for zenith, offset in itertools.product(self.args["zeniths"], self.args["offsets"]):
+            file_name = self.args["file_name_template"].format(zenith=int(zenith))
+            file_path = Path(self.args["base_path"]).joinpath(file_name)
+
+            if not file_path.exists():
+                self.logger.warning(f"File not found: {file_path}. Skipping.")
+                continue
+
+            evaluator = StatisticalErrorEvaluator(
+                file_path,
+                file_type="Gamma-cone",
+                metrics=self.metrics,
+                grid_point=(1 * u.TeV, 180 * u.deg, zenith, 0, offset * u.deg),
+            )
+            evaluator.calculate_metrics()
+            self.evaluator_instances.append(evaluator)
+
+    def perform_interpolation(self):
+        """Perform interpolation for the query point."""
+        if not self.evaluator_instances:
+            self.logger.error("No evaluators initialized. Cannot perform interpolation.")
+            return None
+
+        interpolation_handler = InterpolationHandler(self.evaluator_instances, metrics=self.metrics)
+        query_point = self.args.get("query_point")
+        if not query_point or len(query_point) != 5:
+            raise ValueError(
+                "Invalid query point format. "
+                f"Expected 5 values, got {len(query_point) if query_point else 'None'}."
+            )
+        query_points = np.array([self.args["query_point"]])
+        return interpolation_handler.interpolate(query_points)
+
+    def write_output(self, scaled_events):
+        """Write the interpolation results to a file."""
+        output_data = {
+            "query_point": self.args["query_point"],
+            "scaled_events": scaled_events.tolist(),
+        }
+        with open(self.output_filepath, "w", encoding="utf-8") as f:
+            json.dump(output_data, f, indent=4)
+        self.logger.info(f"Output saved to {self.output_filepath}")
+        self.logger.info(
+            f"Scaled events for grid point {self.args['query_point']}: {scaled_events}"
+        )
+
+    def run(self):
+        """Run the scaling and interpolation workflow."""
+        self.logger.info(f"args dict: {self.args}")
```

Comment:
This is printing a lot of unrelated args_dict to the screen which might be confusing for the user. Do we want to concentrate on those relevant to this application?